analyzere/base_resources.py

```diff
@@ -222,18 +222,28 @@ def upload_status(self):
     def upload_data(self, file_or_str):
         """
         Accepts a file-like object or string and uploads it. Files are
-        automatically uploaded in 4mb chunks. Implements the tus protocol.
+        automatically uploaded in chunks. The default chunk size is 16MiB and
+        can be overwritten by specifying the number of bytes in the
+        ``analyzere.upload_chunk_size`` variable.
+        Implements the tus protocol.
         """
         file_obj = StringIO(file_or_str) if isinstance(
             file_or_str, six.string_types) else file_or_str
-        chunk_size = 4 * 1024 * 1024
-        length = utils.file_length(file_obj)
 
-        # Initiate upload session
-        request_raw('post', self._data_path, headers={'Entity-Length': length})
+        # Upload file with known entity size if file object supports random
+        # access.
+        if hasattr(file_obj, 'seek'):
+            length = utils.file_length(file_obj)
+
+            # Initiate upload session
+            request_raw('post', self._data_path,
+                        headers={'Entity-Length': length})
+        else:
+            request_raw('post', self._data_path)
```

Comment:
by the way, recent versions of the code support TUS 1.0 headers, so maybe it would be good to clean this up to use them now unless for some reason we need it to be backwards compatible with old servers.

it should be mostly a s/Offset/Upload-Offset/  and a s/Entity-Length/Upload-Length/

of course this could be done outside the scope of this PR for sure.